backend/src/services/recurring_charges/criteria_builders.py

```
@@ -0,0 +1,545 @@
+        frequency = TemporalCriteriaBuilder._detect_frequency(avg_interval)
+        
+        # Determine temporal pattern type
+        pattern_type = TemporalPatternType.FLEXIBLE
+        suggested_day = None
+        
+        if day_of_month_variance < 3:  # Consistent day of month
+            pattern_type = TemporalPatternType.DAY_OF_MONTH
+            suggested_day = day_of_month_mode
+        elif day_of_week_consistency > 0.8:  # Consistent day of week
+            pattern_type = TemporalPatternType.DAY_OF_WEEK
+            suggested_day = day_of_week_mode
```

Ruff (0.14.6)

Local variable `suggested_day` is assigned to but never used

Remove assignment to unused variable `suggested_day`

(F841)